### PR TITLE
Implements container.rebind support #484

### PR DESCRIPTION
**Pull request by remojansen**
## Description
Implements container.rebind support

## Related Issue
#484

## Motivation and Context
#484

## How Has This Been Tested?
CI + Additional unit tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jamesadarich** [comment]
@remojansen does this require a review?

**Comment by remojansen** [comment]
Hi @jamesrichford feel free to comment if you see anything ugly 😉

File: src/container/container.ts [hunk 1]
```diff
@@ -108,11 +108,29 @@ class Container implements interfaces.Container {
             };
         };
 
+        let getRebindFunction = (moduleId: string) => {
+            return (serviceIdentifier: interfaces.ServiceIdentifier<any>) => {
+                let _rebind = this.rebind.bind(this);
```
Comment (jamesadarich) [inline src/container/container.ts:113]: Maybe change these two lines to

```
const bindingToSyntax = this.rebind.call(this, serviceIdentifier);
```

As having three word bind three times in a row is a bit of a mind bender! Haha

**Review by jamesadarich — COMMENTED** [review]
Just the one minor thing there. Apart from the tests not being written in Alsatian of course ;)

File: src/container/container.ts (hunk 1 above)
Comment (remojansen) [reply src/container/container.ts:113]: :+1: thanks a lot for the review :100: I have done those changes at https://github.com/inversify/InversifyJS/pull/491

**Pull request merged**